manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -174,64 +187,78 @@ func NewPostgresqlDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, erro
 			corev1.VolumeMount{Name: "miq-pg-configs", MountPath: "/opt/app-root/src/postgresql-cfg/"},
 		},
 	}
+}
 
-	err := addResourceReqs(cr.Spec.PostgresqlMemoryLimit, cr.Spec.PostgresqlMemoryRequest, cr.Spec.PostgresqlCpuLimit, cr.Spec.PostgresqlCpuRequest, &container)
+func postgresqlPodSpec(cr *miqv1alpha1.ManageIQ) (corev1.PodSpec, error) {
+	container := postgresqlContainer(cr)
+	err := addResourceReqs(cr.Spec.PostgresqlMemoryLimit, cr.Spec.PostgresqlMemoryRequest, cr.Spec.PostgresqlCpuRequest, cr.Spec.PostgresqlCpuLimit, &container)
 	if err != nil {
-		return nil, err
+		return corev1.PodSpec{}, err
 	}
+	return corev1.PodSpec{
+		Containers: []corev1.Container{
+			container,
+		},
+		Volumes: []corev1.Volume{
+			corev1.Volume{
+				Name: "miq-pgdb-volume",
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: "postgresql",
+					},
+				},
+			},
+			corev1.Volume{
+				Name: "miq-pg-configs",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: "postgresql-configs"},
+					},
+				},
+			},
+		},
+	}, nil
+
+}
+
+func NewPostgresqlDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, controllerutil.MutateFn, error) {
 
-	deploymentLabels := map[string]string{
-		"app": cr.Spec.AppName,
-	}
 	podLabels := map[string]string{
 		"name": "postgresql",
 		"app":  cr.Spec.AppName,
 	}
+
+	podSpec, err := postgresqlPodSpec(cr)
+
+	if err != nil {
+		return nil, nil, err
+	}
 	var repNum int32 = 1
 
 	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "postgresql",
 			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    deploymentLabels,
+			Labels:    podLabels,
 		},
 		Spec: appsv1.DeploymentSpec{
-			Strategy: appsv1.DeploymentStrategy{
-				Type: "Recreate",
-			},
-			Replicas: &repNum,
 			Selector: &metav1.LabelSelector{
 				MatchLabels: podLabels,
```

Comment:
I think we should be ensuring that this is correct in the mutate function. Can you model this after [the way we did it for memcached](https://github.com/ManageIQ/manageiq-pods/blob/0e3e06e2ac81923b8fb27064b1eef40b599d1d54/manageiq-operator/pkg/helpers/miq-components/memcached.go#L81)?